src/test/java/testapp/adapter/http/UserServiceTest.java

```diff
@@ -0,0 +1,164 @@
+package testapp.adapter.http;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalMemcacheServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sitebricks.client.transport.Json;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.headless.Request;
+import org.jmock.Expectations;
+import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import testapp.adapter.listeners.UserLogListener;
+import testapp.adapter.repository.UserPersistentRepository;
+import testapp.core.FakeRequestRead;
+import testapp.core.User;
+import testapp.core.UserRepository;
+import testapp.core.UserValidator;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertThat;
+import static testapp.matchers.UserMatchers.returns;
+
+public class UserServiceTest {
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+    private LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalMemcacheServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+    @Before
+    public void setUp() throws Exception {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        helper.tearDown();
+    }
+
+    @Test
+    public void registerUser() throws Exception {
+        final UserRepository repository = new UserValidator(new UserPersistentRepository(new UserLogListener()));
```

Comment:
It would be better to instantiate Guice module to avoid these repeating instantiates(if you want to keep the end2end tests)